### PR TITLE
Connect register route and add bcrypt auth mahi

### LINKED ISSUE #36: create POST /api/v1/auth/register route
[Auth](https://github.com/harakeke-2019/jwt-auth)

Create a server/db/users.js that has a createUser db function.
Follow the rest of the instructions linked above.

Update the [wiki](https://github.com/harakeke-2019/EDAMAME/wiki/Server-Routes) accordingly.

### PR DESCRIPTION
**Pull request by benlittlenz**
closes #36 
Kia ora we need this to progress with connecting to the registration form. 

thanks :)

### CODE REVIEW

**Review by chrislaupama — DISMISSED** [review]
Awesome work team!

**Comment by benlittlenz** [comment]
Kia ora @chrislaupama we have pushed up more stuff for review. Thanks!

File: package.json [hunk 1]
```diff
@@ -8,7 +8,7 @@
   "license": "ISC",
   "scripts": {
     "start": "webpack && node server",
-    "dev": "node server & webpack --watch",
+    "dev": "nodemon server & webpack --watch",
```
Comment (emilyparkes) [inline package.json:11]: Please remove this change as it is irrelevant to your ticket. If this change is needed create a new ticket please.

File: server/routes/starter.js [hunk 2]
```diff
@@ -1,11 +1,9 @@
 const express = require('express')
-//const db = require('../../db')
+// const db = require('../../db')
```
Comment (emilyparkes) [inline server/routes/starter.js:2]: Why was this change made?

File: server/server.js [hunk 3]
```diff
@@ -1,17 +1,20 @@
 const path = require('path')
 const express = require('express')
-
 const server = express()
 
 const starterRoute = require('./routes/starter')
 const usersRoutes = require('./routes/usersRoutes')
 const cohortRoutes = require('./routes/cohortRoutes')
+const auth = require('./routes/auth')
```
Comment (emilyparkes) [inline server/server.js:8]: Please follow code style of `const authRoutes = require('./routes/auth')`

File: server/server.js [hunk 4]
```diff
@@ -1,17 +1,20 @@
...
+const users = require('./routes/users')
 
 server.use(express.json())
 server.use(express.static(path.join(__dirname, './public')))
 
+server.use('/api/v1/users', users)
 server.use('/', starterRoute)
 server.use('/api/v1/users', usersRoutes)
 server.use('/api/v1/cohorts', cohortRoutes)
+server.use('/api/v1/auth/register', auth)
```
Comment (emilyparkes) [inline server/server.js:20]: Please follow code style of `server.use('/api/v1/auth/register', authRoutes)`

**Review by emilyparkes — CHANGES_REQUESTED** [review]

**Comment by benlittlenz** [comment]
Closed without merging because we will no longer use bcrypt for authentication

**Pull request closed without merging**